### DISCUSSION TITLE
Confusing `require_root` warning message for brew services, inconsistent with brew's anti-root stance (which I like!)

### DISCUSSION THREAD

**Discussion by sandstrom** [Homebrew/discussions#4215] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.21
ORIGIN: https://github.com/Homebrew/brew
HEAD: 932d2cf3b77c9439a57b6a43577fc8d3b6399a62
Last commit: 8 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e80ba2c11dc2900ada9cc90bb2c97b7ddf4a14bc
Core tap last commit: 3 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nano
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_ENV_HINTS: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 13.0.0 build 1300
Git: 2.37.3 => /usr/local/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2-x86_64
CLT: N/A
Xcode: 13.2.1
```

### Output of `brew doctor`

```shell
Warning: Your Xcode (13.2.1) is too outdated.
Please update to Xcode 14.1 (or delete it).
Xcode can be updated from the App Store.

Warning: Your Xcode (13.2.1) is outdated.
Please update to Xcode 14.1 (or delete it).
Xcode can be updated from the App Store.
```

### Description of issue

The following warning from homebrew is misleading:

> % brew services start dnsmasq
> Warning: dnsmasq must be run as root to start at system startup!
> ==> Successfully started `dnsmasq` (label: homebrew.mxcl.dnsmasq)

According to brew services docks, things will startup even though they aren't setup with root. The only difference is that they'll startup at "login" instead of "boot". In some cases this difference is important, but in others it isn't.

From homebrew services docs:

> % brew services --help
> Usage: brew services [subcommand]
> 
> Manage background services with macOS' launchctl(1) daemon manager.
> 
> If sudo is passed, operate on /Library/LaunchDaemons (started at boot).
> Otherwise, operate on ~/Library/LaunchAgents (started at login).

For our use-case, we prefer to run dnsmasq as a non-root user (we run it on 5353), and it works great. But I was thrown off for a long time, thinking that it wasn't possible to have it auto-start unless I used `sudo brew services …` (and I know sudo is generally discouraged).

So my thinking is, maybe the `require_root` warning should be removed?

https://github.com/Homebrew/homebrew-core/blob/master/Formula/dnsmasq.rb#L66

Or changed to something like this:

> When dnsmasq isn't run as root, it will be started at user login instead of system startup. For some cases this doesn't matter, but in case it does, consider using root instead.

**Comment by SMillerDev** [comment]
> When dnsmasq isn't run as root, it will be started at user login instead of system startup. For some cases this doesn't matter, but in case it does, consider using root instead.

That just makes it a lot more vague if it'll work or not. In the default configuration dnsmasq needs to be started as root, it's that simple.

**Reply by sandstrom** [reply]
@SMillerDev Because of binding to port 53, which requires root? (I know that's required for 53, but is that the reason you are referring too)

**Reply by SMillerDev** [reply]
That's the main one, yes

**Reply by sandstrom** [reply]
It's a good point, I agree with it.

In the case of `dnsmasq` (and likely other services too), removing that message would make more harm.

Within the APIs of brew, and the spec for a formula, is there maybe another way of somehow clarifying that brew services can (at least sometimes) be used without `sudo`?

Especially since that's what brew in general encourages!

Maybe I'm just stupid, but I was somewhat thrown off by those messages and it took some digging to understand how the sudo/non-sudo services bootup stuff worked.

Maybe it's possible to strike a balance between the two, somehow?
